src/main/comments.js

```diff
@@ -583,10 +578,20 @@ function printComments(path, printed, options, needsSemi) {
     }
   }, "comments");
 
-  return prependCursorPlaceholder(path, options, [
-    ...leadingParts,
-    ...trailingParts,
-  ]);
+  if (isCursorNode) {
+    leadingParts.unshift(cursor);
+    trailingParts.push(cursor);
+  }
+
+  return { leading: leadingParts, trailing: trailingParts };
+}
+
+function printComments(path, doc, options, needsSemi) {
```

Comment:
As I can see `needsSemi` is not used at all, and I don't think plugin use it, lets remove it.

Reply:
Oh, just one place `printStatementSequence` use it, but it can print semi itself.

Reply:
Done!

Reply:
Can't it done here https://github.com/prettier/prettier/blob/863eb079dc0aa6478b24842f69bef2b99408c948/src/language-js/print/statement.js#L49 ?

Reply:
In that case we will need a way to say `print` that it shouldn't print comments. How would you do that?

Reply:
You are right.

Reply:
After second thought, we can delete comments then print without comments :smile: But it still won't work, because statements may need parens, semicolon will be inside.